cidc_schemas/schemas/assays/components/ngs/ngs_assay_record.json

```
@@ -11,7 +11,17 @@
     },
     "files": {
       "$comment": "Fastq and read group mapping files produced by the assay for that .",
-      "$ref": "assays/components/fastq_pair.json"
+      "anyOf": [
+        {
+            "$ref": "assays/components/fastq_pair.json"
+        },
+        {
+            "bam": {
```

After more though I think this is not what you want. I think this `bam` key doesn't get validated at all - i think it's ignored by schema. The way you want it is probably something like 
```suggestion
            "properties": {
                "bam": {
```

then it will be checked.

---

resolved by your PR